src/utils.h

```diff
@@ -122,6 +122,19 @@ memdup(const void *mem, size_t nmemb, size_t size)
     return p;
 }
 
+#ifdef _MSC_VER
+static inline char *
+strndup(const char *str, size_t len)
+{
+    char *p = calloc(len + 1, sizeof(char));
```

Comment:
Would be better to do a `strlen(s)` and take the min instead of always allocating `len+1` bytes. E.g. `strndup("foo", 10000)` should only allocate 4 bytes.

Reply:
Indeed, thanks for the tip!